test/Microsoft.AspNetCore.Mvc.Core.Test/ModelBinding/Binders/ComplexTypeModelBinderTest.cs

```diff
@@ -350,6 +350,52 @@ public void CreateModel_InstantiatesInstanceOfMetadataType()
         }
 
         [Fact]
+        public void CreateModel_ForStructModelType_AsTopLevelObject_ThrowsException()
+        {
+            // Arrange
+            var bindingContext = new DefaultModelBindingContext
+            {
+                ModelMetadata = GetMetadataForType(typeof(PointStruct)),
+                IsTopLevelObject = true
+            };
+            var binder = CreateBinder(bindingContext.ModelMetadata);
+
+            // Act & Assert
+            var exception = Assert.Throws<InvalidOperationException>(() => binder.CreateModelPublic(bindingContext));
+            Assert.Equal(
+                string.Format(
+                    "Could not create an instance of type '{0}'. Model bound complex types must not be abstract or " +
+                    "value types and must have a parameterless constructor.",
+                    typeof(PointStruct).FullName),
+                exception.Message);
+        }
```

Comment:
Ok that's good. I wasn't super familiar with the commit before this one, but I wanted to ask just to make sure 👍